sys/tsrb/tsrb.c

```diff
@@ -49,6 +49,16 @@ int tsrb_get(tsrb_t *rb, char *dst, size_t n)
     return (n - tmp);
 }
 
+int tsrb_drop(tsrb_t *rb, size_t n)
```

Comment:
how would it look if this would use tsrb_get_one() in a loop?

Reply:
Something like:

```
size_t tmp = n;
while (tmp && (tsrb_get_one(rb) != -1)) {
    tmp--;
}
return (n - tmp);
```

But I'm not sure whether `tsrb_get_one` is bug free. What happens if `_pop` returns -1 because that's what's in the buffer? Is that sign extended and thus interpreted as if the buffer is empty? (Do you want an issue for this?)

Reply:
> What happens if _pop returns -1 because that's what's in the buffer?

That is indeed a serious bug. Seems like slipdev is the only user of tsrb_get_one(), but how could that ever work? :)
We might have to change the whole API to use unsigned char.